editor/components/document-outline/test/index.js

```diff
@@ -74,4 +76,13 @@ describe( 'DocumentOutline', () => {
 			expect( wrapper ).toMatchSnapshot();
 		} );
 	} );
+
+	describe( 'nested headings', () => {
+		it( 'should render even if the heading is nested', () => {
+			const blocks = [ nestedHeading ];
+			const wrapper = shallow( <DocumentOutline blocks={ blocks } /> );
+
+			expect( wrapper.contains( 'Heading child' ) ).toBeTruthy();
+		} );
+	} );
```

Comment:
Maybe a slightly more complex test would be a good complement.

Reply:
The test case was extended 👍